### PR TITLE
Generic improvements (part 1)

### PR DESCRIPTION
**Pull request by oshtaier**
```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_activationkey.py -k 'test_positive_add_rh_product or test_positive_fetch_product_content'
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-18 13:23:39 - conftest - DEBUG - BZ deselect is disabled in settings

collected 28 items / 26 deselected                                                                                                                                                           

tests/foreman/ui/test_activationkey.py ..                                                                                                                                              [100%]
=================================================================== 2 passed, 26 deselected, 5 warnings in 595.82 seconds ==================================================================
```

```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_architecture.py 
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-18 13:40:59 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui/test_architecture.py .                                                                                                                                                [100%]
=========================================================================== 1 passed, 2 warnings in 116.22 seconds ==========================================================================
```

```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_contentview.py -k 'test_positive_end_to_end or test_positive_publish_version_changes_in_source_env or test_positive_create_composite or test_positive_add_synced_docker_repo or test_positive_publish_composite_with_custom_content or test_positive_add_errata_filter or test_positive_add_puppet_module'
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-18 15:15:42 - conftest - DEBUG - BZ deselect is disabled in settings

collected 80 items / 73 deselected                                                                                                                                                           

tests/foreman/ui/test_contentview.py .......                                                                                                                                           [100%]
================================================================== 7 passed, 73 deselected, 10 warnings in 1724.38 seconds ===================================================================
```

```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_computeprofiles.py 
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-18 15:56:11 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui/test_computeprofiles.py .                                                                                                                                             [100%]
=========================================================================== 1 passed, 2 warnings in 106.56 seconds =========================================================================
```

```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_domain.py 
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-18 16:35:32 - conftest - DEBUG - BZ deselect is disabled in settings

collected 6 items                                                                                                                                                                            

tests/foreman/ui/test_domain.py ......                                                                                                                                                 [100%]
=========================================================================== 6 passed, 2 warnings in 553.59 seconds =======================================================================
```

```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_lifecycleenvironment.py::test_positive_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-18 18:07:50 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui/test_lifecycleenvironment.py .                                                                                                                                        [100%]
=========================================================================== 1 passed, 9 warnings in 96.21 seconds ============================================================================
```

```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_operatingsystem.py::test_positive_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-19 14:42:34 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui/test_operatingsystem.py .                                                                                                                                             [100%]
=========================================================================== 1 passed, 2 warnings in 219.39 seconds ====================================================================
```

```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_usergroup.py::test_positive_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-19 15:01:42 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui/test_usergroup.py .                                                                                                                                                   [100%]
=========================================================================== 1 passed, 2 warnings in 106.85 seconds =========================================================================
```

```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_repository.py -k 'test_positive_end_to_end_custom_yum_crud or test_positive_discover_repo_via_new_product'
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-19 15:53:17 - conftest - DEBUG - BZ deselect is disabled in settings

collected 20 items / 18 deselected                                                                                                                                                           

tests/foreman/ui/test_repository.py ..                                                                                                                                                 [100%]
=================================================================== 2 passed, 18 deselected, 11 warnings in 301.78 seconds ================================================================
```

```
py.test /home/ashtayer/Desktop/robottelo/tests/foreman/ui/test_jobtemplate.py::test_positive_end_to_end
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.6.5, pytest-4.0.2, py-1.5.3, pluggy-0.11.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ashtayer/Desktop/robottelo, inifile:
plugins: services-1.3.1, mock-1.10.0
collecting ... 2019-06-19 16:05:24 - conftest - DEBUG - BZ deselect is disabled in settings

collected 1 item                                                                                                                                                                             

tests/foreman/ui/test_jobtemplate.py .                                                                                                                                                 [100%]
========================================================================== 1 passed, 27 warnings in 197.78 seconds ===========================================================================
```

### CODE REVIEW

File: airgun/entities/domain.py [hunk 1]
```diff
@@ -51,27 +57,32 @@ def add_parameter(self, entity_name, param_name, param_value):
...
         view.flash.assert_no_error()
         view.flash.dismiss()
 
-    def delete(self, name):
+    def delete(self, entity_name):
+        """Delete existing domain entity"""
         view = self.navigate_to(self, 'All')
-        self.search(name)
-        if not view.table.row_count:
-            raise ValueError("Unable to find name '{}'".format(name))
```
[the remark this answers is not in the document]
Comment (oshtaier) [reply airgun/entities/domain.py]: I believe that entity was implemented first and search method was very primitive. It did not produce proper exception if no rows returned. We already made that idea generic for a long time.

[the remark this answers is not in the document]
Comment (oshtaier) [reply airgun/entities/domain.py]: So yeah, it is completely redundant

File: airgun/views/contentview.py [hunk 2]
```diff
@@ -132,7 +132,7 @@ class details(SatTab):
     @View.nested
     class versions(SatTab):
         searchbox = Search()
-        table = SatTable(
+        table = Table(
             locator='//table',
             column_widgets={
```
Comment (omkarkhatavkar) [inline airgun/views/contentview.py]: any issues with './/table' ?

File: airgun/views/contentview.py [hunk 3]
```diff
@@ -179,7 +179,7 @@ class filters(SatTabWithDropdown, SearchableViewMixin):
         new_filter = Text(".//button[@ui-sref='content-view.yum.filters.new']")
         remove_selected = Text(".//button[@ng-click='removeFilters()']")
 
-        table = SatTable(
+        table = Table(
             locator='//table',
             column_widgets={
```
Comment (omkarkhatavkar) [inline airgun/views/contentview.py]: any issues with './/table' ?

File: airgun/views/contentview.py (hunk 2 above)
Comment (oshtaier) [reply airgun/views/contentview.py]: Maybe, no issues, but that was not intention of current PR. I can look on that in next improvements

File: airgun/views/contentview.py (hunk 3 above)
Comment (oshtaier) [reply airgun/views/contentview.py]: same

File: airgun/views/contentview.py (hunk 2 above)
Comment (omkarkhatavkar) [reply airgun/views/contentview.py]: Ok NP

**Comment by oshtaier** [comment]
feel free to merge if you okay with PR, so I can move with updated code in my work

**Comment by omkarkhatavkar** [comment]
> if you okay with PR, so I can move with updated code in my work

sure I will

**Pull request merged**